src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,28 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        UserData cachedData = cache.get(userId);
+        if (cachedData != null) {
+            return CompletableFuture.completedFuture(cachedData);
+        }
+        return CompletableFuture
+                .supplyAsync(() -> new UserData(userId, "Details for " + userId), executor)
+                .thenApply(userData -> {
+                    cache.putIfAbsent(userId, userData);
+                    return userData;
+                });
```

Comment:
While the method 'processRequest' correctly checks for cached data and uses 'supplyAsync' to process the request asynchronously, there is a potential race condition. If concurrent requests for the same 'userId' occur before the cache is updated, multiple UserData objects could be created and processed, which is likely not the intended behavior. To ensure thread safety and avoid unnecessary processing, consider adding a mechanism to handle concurrent requests for the same 'userId' more efficiently.